x-pack/plugins/monitoring_collection/README.md

```diff
@@ -2,4 +2,111 @@
 
 ## Plugin
 
-This plugin allows for other plugins to add data to Kibana stack monitoring documents.
\ No newline at end of file
+This plugin allows for other plugins to add data to Kibana stack monitoring documents.
+
+## OpenTelemetry Metrics
+
+### Enable Prometheus endpoint with Elastic Agent Prometheus input
+
+1. Start [local setup with fleet](../fleet/README.md#running-fleet-server-locally-in-a-container) or a cloud cluster
+2. Start Kibana
+3. Set up a new agent policy and enroll a new agent in your local machine
+4. Install the Prometheus Metrics package
+  a. Set **Hosts** with `localhost:5601`
+  b. Set **Metrics Path** with `/(BASEPATH)/api/monitoring_collection/v1/prometheus`
+  c. Remove the values from **Bearer Token File** and **SSL Certificate Authorities**
+  d. Set **Username** and **Password** with `elastic` and `changeme`
+5. Add the following configuration to `kibana.dev.yml`
+
+    ```yml
+    # Enable the prometheus exporter
+    monitoring_collection.opentelemetry.metrics:
+      prometheus.enabled: true
+
+    ```
+
+### Enable OpenTelemetry Metrics API exported as OpenTelemetry Protocol
+
+1. Start [local setup with fleet](../fleet/README.md#running-fleet-server-locally-in-a-container) or a cloud cluster
+2. Start Kibana
+3. Set up a new agent policy and enroll a new agent in your local machine
+4. Install Elastic APM package listening on `localhost:8200` without authentication
+5. Add the following configuration to `kibana.dev.yml`
+
+    ```yml
+    # Enable the OTLP exporter
+    monitoring_collection.opentelemetry.metrics:
+      otlp.url: "http://127.0.0.1:8200"
+    ```
+
+### Example of how to instrument the code
+
+* First, we need to define what metrics we want to instrument with OpenTelemetry
+
+  ```ts
+  import { Counter, Meter } from '@opentelemetry/api-metrics';
+
+  export class FooApiMeters {
+    requestCount: Counter;
+
+    constructor(meter: Meter) {
+      this.requestCount = meter.createCounter('request_count', {
+        description: 'Counts total number of requests',
+      });
+    }
+  ```
+
+  In this example we're using a `Counter` metric, but [OpenTelemetry SDK](https://open-telemetry.github.io/opentelemetry-js/interfaces/_opentelemetry_api_metrics.Meter.html) provides there are other options to record metrics
```

Comment:
Not sure why the code hlighting goes weird here. Will see if I can figure it out.

Reply:
What did you mean by this?